test/gtest/gathernd.cpp

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "gathernd.hpp"
+
+using GPU_GatherND_bwd_FP32_UINT32  = GatherNDBwdTest<float, uint32_t>;
+using GPU_GatherND_bwd_FP16_UINT32  = GatherNDBwdTest<half, uint32_t>;
+using GPU_GatherND_bwd_BFP16_UINT32 = GatherNDBwdTest<bfloat16, uint32_t>;
+using GPU_GatherND_bwd_FP32_UINT64  = GatherNDBwdTest<float, uint64_t>;
+using GPU_GatherND_bwd_FP16_UINT64  = GatherNDBwdTest<half, uint64_t>;
+using GPU_GatherND_bwd_BFP16_UINT64 = GatherNDBwdTest<bfloat16, uint64_t>;
```

Comment:
I think you should consider generalizing the indices' data type here. If you intend to support both `uint32_t` and `uint64_t`, let's just simply use `uint64_t` since it can handle both cases (supporting `uint64_t` inherently covers `uint32_t`)

Reply:
hmmm i check another pytorch op and realized that they actually separate the implementation of `int32_t` and `int64_t`, but im not sure about the reason why

Reply:
cause in some cases, you don't need 64 bit for `indices` and using only 32 bit would reduce half of memory needed for storing `indices`